src/main/cpp/RobotContainer.cpp

```diff
@@ -518,59 +545,110 @@ frc2::CommandPtr RobotContainer::GetAutonomousCommand()
           frc2::WaitCommand(0.5_s).ToPtr() //can change if need
         )
       ),
-      AutoShootCommand(m_shooterWheels, m_intake).ToPtr(),
+      AutoShootCommand(m_shooterWheels, m_intake).ToPtr()
+      // frc2::cmd::RunOnce(
+      //   [this]
+      //   {
+      //     m_drive.ZeroHeading(m_drive.GetPose().Rotation().Degrees());
+      //   },
+      //   {&m_drive}
+      // )
+    );  
+  }
+  */
+  else if(chosenAuto == "Close4")
+  {
+    m_drive.ResetOdometry(close4Waypoint1[0]);
+    return frc2::cmd::Sequence( //the whole auto path!
+      frc2::WaitCommand(0.1_s).ToPtr(),  //This is neccesary because the reset odometry will not actually reset until after a very small amount of time. 
+      AutoShooterWarmupCmd(m_shooterWheels).ToPtr(),
+      frc2::cmd::Race(
+        AutoSubAim(m_shooter).ToPtr(),
+        frc2::cmd::Sequence(
+          frc2::WaitCommand(0.75_s).ToPtr(), //can change if need
+          AutoShootCommand(m_shooterWheels, m_intake).ToPtr())
+      ),
+      frc2::cmd::Parallel(
+        FollowWaypoints(m_drive, m_limelight, close4Waypoint1, close4PointSpeed1, close4CruiseSpeed1, false).ToPtr(),
+        IntakeCmd(m_intake).ToPtr()
+      ),
+      frc2::cmd::Race( //aim shooter for 0.75s
+        AutoAprilTag(m_limelight, m_drive, m_shooter).ToPtr(),
+        frc2::cmd::Sequence(
+          frc2::WaitCommand(0.75_s).ToPtr(), //can change if need
+          AutoShootCommand(m_shooterWheels, m_intake).ToPtr())
+      ),
+      frc2::cmd::Parallel(
+        FollowWaypoints(m_drive, m_limelight, close4Waypoint2, close4PointSpeed2, close4CruiseSpeed2, false).ToPtr(),
+        IntakeCmd(m_intake).ToPtr()
+      ),
+      frc2::cmd::Race( //aim shooter for 0.75s
+        AutoAprilTag(m_limelight, m_drive, m_shooter).ToPtr(),
+        frc2::cmd::Sequence(
+          frc2::WaitCommand(0.75_s).ToPtr(), //can change if need
+          AutoShootCommand(m_shooterWheels, m_intake).ToPtr())
+      ),
+      frc2::cmd::Parallel(
+        FollowWaypoints(m_drive, m_limelight, close4Waypoint3, close4PointSpeed3, close4CruiseSpeed3, false).ToPtr(),
+        IntakeCmd(m_intake).ToPtr()
+      ),
+      frc2::cmd::Race( //aim shooter for 0.75s
+        AutoAprilTag(m_limelight, m_drive, m_shooter).ToPtr(),
+        frc2::cmd::Sequence(
+          frc2::WaitCommand(0.75_s).ToPtr(), //can change if need
+          AutoShootCommand(m_shooterWheels, m_intake).ToPtr())
+      ),
       frc2::cmd::RunOnce(
         [this]
         {
           m_drive.ZeroHeading(m_drive.GetPose().Rotation().Degrees());
```

Comment:
instead of having this at the very end it could be in teleop init. Would have a lot less duplicate code

Reply:
I think I mentioned this somewhere but in the beginning of auto basically guarantees your rotation is the same each time, so zeroing heading is working properly. Otherwise the robot ends up who knows where at the end of auto and then it doesn't work the same way to properly adjust the heading. You can try and get it to work but I think your heading will be off if you wait until tele op.